### ISSUE TITLE
Filtering broken for Combobox/Autocomplete with itemToString

### ISSUE THREAD

**Issue by Procrat** [segmentio/evergreen#930]
#### Summary
Filtering doesn't work for a `Combobox` (and presumably also `Autocomplete`) with `itemToString`.

#### Steps to reproduce
```
import React from 'react';
import { Combobox } from 'evergreen-ui';

export default () => (
  <Combobox
    items={[{label: 'label'}]}
    itemToString={item => item ? item.label : ''}
  />
);
```

#### Expected behaviour
When typing in a substring of "label", I expect the "label" item to show.

#### Observed behaviour
It doesn't.

#### Underlying issue
The V5 commit (731e46e529030d8062deba833c42ebeaf9066f3f) moved the [assignment to `itemsFilter` around to a place where `itemToString` isn't defined yet](https://github.com/segmentio/evergreen/blob/04855bcb02c9e0b1e3e37e041d573e0a6f189734/src/autocomplete/src/Autocomplete.js#L46). It has an obvious fix, so I'll make a PR for it.

**Comment by colinlohner** [comment]
@Procrat what version of Evergreen where you using when you saw this issue?

We also saw this issue internally and found that it was actually being caused by the itemToString prop not being passed down to the Downshift component. We resolved it in this PR: https://github.com/segmentio/evergreen/pull/924/files and it was released in Evergreen v5.0.2

Looking at your PR, it looks like what you are doing is actually going to have the same results as what is there currently. Happy to answer any questions you may have

**Comment by colinlohner** [comment]
Good catch! Thanks for the PR as well.

**Comment by colinlohner** [comment]
Released you can now use as you were and the positioning will be handled.

Available in v5.0.4
